# DANS CE PROJET NOUS ALLONS ETUDIER LES DONNEES DE LA BANQUE MONDIALE. 
# EN PARTICULIER LES DONNEES STATISTIQUES SUR L'EDUCATION. 


In [305]:
# COMMENCONS PAR IMPORTER LES LIBRAIRIES NECESSAIRES 
import numpy as np # POUR LE CALCUL MATRICIEL NOTAMMENT
import pandas as pd # POUR LES DATAFRAMES
import matplotlib as plt # POUR LES REPRESENTATIONS GRAPHIQUES
import seaborn as sns # pour une plus grande facilité à réaliser des graphiques "attrayants"

### DANS LE DOSSIER A TELECHARGER SE TROUVENT 5 FICHIERS .CSV. NOUS ALLONS LES ETUDIER UN PAR UN

# ***<u>ETUDE DU PREMIER FICHIER : EdStatsCountry</u>***

In [306]:
# definition du path
path=r'C:\Users\lenovo\Desktop\OpenClassrooms\PROJET_OC\PROJET_2\RESSOURCES\Projet+Python_Dataset_Edstats_csv\EdStatsCountry.csv'

In [307]:
# CREATION DU DATAFRAME CORRESPONDANT
Fichier_Country = pd.read_csv(path)

In [308]:
# ETUDE DES 5 PREMIERES LIGNES DU DATAFRAME
Fichier_Country.head(2)

Country Code   Short Name   Table Name                     Long Name  \
0          ABW        Aruba        Aruba                         Aruba   
1          AFG  Afghanistan  Afghanistan  Islamic State of Afghanistan   

  2-alpha code   Currency Unit  \
0           AW   Aruban florin   
1           AF  Afghan afghani   

                                       Special Notes  \
0  SNA data for 2000-2011 are updated from offici...   
1  Fiscal year end: March 20; reporting period fo...   

                      Region          Income Group WB-2 code  ...  \
0  Latin America & Caribbean  High income: nonOECD        AW  ...   
1                 South Asia            Low income        AF  ...   

            IMF data dissemination standard  Latest population census  \
0                                       NaN                      2010   
1  General Data Dissemination System (GDDS)                      1979   

                             Latest household survey  \
0                                                NaN   
1  Multiple Indicator Cluster Survey (MICS), 2010/11   

  Source of most recent Income and expenditure data  \
0                                               NaN   
1           Integrated household survey (IHS), 2008   

  Vital registration complete Latest agricultural census  \
0                         Yes                        NaN   
1                         NaN                    2013/14   

  Latest industrial data Latest trade data Latest water withdrawal data  \
0                    NaN            2012.0                          NaN   
1                    NaN            2012.0                         2000   

  Unnamed: 31  
0         NaN  
1         NaN  

[2 rows x 32 columns]

DE PRIME ABORD ON PEUT CONSTATER QUE CE FICHIER EST TRIE PAR ORDRE ALPHABETIQUE DE COUNTRY CODE

IL COMPORTE 32 COLONNES COMME LE COUNTRY CODE QUI SERT D'IDENTIFIANT

LES DIFFERENTS NOMS (SHORT, TABLE, LONG)

LE PIB (HIGH, LOW ...)

DES DATES DE RECENSEMENT DE LA POPULATION ET DES PARCELLES AGRICOLES ET AUTRES


In [309]:
# RENTRONS PLUS EN DETAIL DANS L'EXPLORATION DE CE DATAFRAME
Fichier_Country.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 241 entries, 0 to 240
Data columns (total 32 columns):
 #   Column                                             Non-Null Count  Dtype  
---  ------                                             --------------  -----  
 0   Country Code                                       241 non-null    object 
 1   Short Name                                         241 non-null    object 
 2   Table Name                                         241 non-null    object 
 3   Long Name                                          241 non-null    object 
 4   2-alpha code                                       238 non-null    object 
 5   Currency Unit                                      215 non-null    object 
 6   Special Notes                                      145 non-null    object 
 7   Region                                             214 non-null    object 
 8   Income Group                                       214 non-null    object 
 9   WB-2 code 

CE FICHIER CONTIENT  241 LIGNES NUMEROTEES DE 0 A 240 ET 32 COLONNES NUMEROTEES DE 0 A 31. IL COMPORTE 4 COLONNES DE TYPES FLOAT (NOMBRES DECIMAUX) ET 28 DE TYPE OBJECT (CHAINE DE CARACTERES). 

ON CONSTATE QUE LA COLONNE N° 31 (Unnamed: 31) EST ENTIEREMENT VIDE, EN EFFET ELLE CONTIENT 0 NON-NULL. D'AUTRES COMME LA N°16 (Alternative conversion factor) OU ENCORE LA N°11 (National accounts reference year).... NE SONT QUE PEU REMPLIES. D'AUTRES COMME LA N°5 (Currency Unit) SONT QUASIMENT COMPLETES. ENFIN DES COLONNES TELLES QUE LA N°1 (Short Name) sont complétes. 

In [310]:
# ON VA DROP TOUTES LES COLONNES QUI N'ONT PAS AU MOINS 70% DE NON NULL

Liste_à_drop = []
for features in Fichier_Country.columns:
    if Fichier_Country[features].isnull().sum()>0.3*Fichier_Country.shape[0]:
        Liste_à_drop.append(features)
print(Liste_à_drop)
Fichier_Country=Fichier_Country.drop(Liste_à_drop,axis=1)
Fichier_Country.info()

['Special Notes', 'National accounts reference year', 'Lending category', 'Other groups', 'Alternative conversion factor', 'PPP survey year', 'External debt Reporting status', 'Government Accounting concept', 'Latest household survey', 'Source of most recent Income and expenditure data', 'Vital registration complete', 'Latest agricultural census', 'Latest industrial data', 'Unnamed: 31']
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 241 entries, 0 to 240
Data columns (total 18 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   Country Code                       241 non-null    object 
 1   Short Name                         241 non-null    object 
 2   Table Name                         241 non-null    object 
 3   Long Name                          241 non-null    object 
 4   2-alpha code                       238 non-null    object 
 5   Currency Unit                      215 non-null    

# ***<u>PASSONS AU PROCHAIN FICHIER NOMME EdStatsCountry-Series</u>***


In [311]:
# definition du path
path=r'C:\Users\lenovo\Desktop\OpenClassrooms\PROJET_OC\PROJET_2\RESSOURCES\Projet+Python_Dataset_Edstats_csv\EdStatsCountry-Series.csv'

In [312]:
# CREATION DU DATAFRAME CORRESPONDANT
Fichier_Country_Series = pd.read_csv(path)

In [313]:
# ETUDE DES 5 PREMIERES LIGNES DU DATAFRAME
Fichier_Country_Series.head(5)

CountryCode         SeriesCode  \
0         ABW        SP.POP.TOTL   
1         ABW        SP.POP.GROW   
2         AFG        SP.POP.GROW   
3         AFG  NY.GDP.PCAP.PP.CD   
4         AFG        SP.POP.TOTL   

                                         DESCRIPTION  Unnamed: 3  
0  Data sources : United Nations World Population...         NaN  
1  Data sources: United Nations World Population ...         NaN  
2  Data sources: United Nations World Population ...         NaN  
3                 Estimates are based on regression.         NaN  
4  Data sources : United Nations World Population...         NaN

DE PRIME ABORD ON PEUT CONSTATER QUE CE FICHIER EST TRIE PAR ORDRE ALPHABETIQUE DE COUNTRY CODE

IL COMPORTE 4 COLONNES COMME LE COUNTRY CODE QUI SERT D'IDENTIFIANT

UNE COLONNE NOMMEE SeriesCode EN LIEN CERTAINEMENT AVEFC LE FICHIER EdStatsSeries

UNE COLONNE DESCRIPTION INDIQUANT LA SOURCE DES INFOS. ON CONSTATE QUE LES SOURCES VARIENT. CERTAINES OFFICIELLES COMME L'ONU. D'AUTRES OBTENUES PAR REGRESSION

UNE COLONNE Unnamed: 3 CERTAINEMENT ENTIEREMENT VIDE COMME SUR LE FICHIER PRECEDENT

In [314]:
# RENTRONS PLUS EN DETAIL DANS L'EXPLORATION DE CE DATAFRAME
Fichier_Country_Series.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 613 entries, 0 to 612
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   CountryCode  613 non-null    object 
 1   SeriesCode   613 non-null    object 
 2   DESCRIPTION  613 non-null    object 
 3   Unnamed: 3   0 non-null      float64
dtypes: float64(1), object(3)
memory usage: 19.3+ KB


CE FICHIER CONTIENT  ***613 LIGNES*** NUMEROTEES DE 0 A 612 ET ***4 COLONNES*** NUMEROTEES DE 0 A 3. IL COMPORTE *****1 COLONNE DE TYPES FLOAT***** (NOMBRES DECIMAUX) ET ***3 DE TYPE OBJECT*** (CHAINE DE CARACTERES).

ON CONSTATE QUE LA COLONNE N° 3 (Unnamed: 3) EST ENTIEREMENT VIDE, EN EFFET ELLE CONTIENT 0 NON-NULL. QUANT AU RESTE DES COLONNES ELLES SONT COMPLETES. 

In [315]:
# ON VA DROP TOUTES LES COLONNES QUI N'ONT PAS AU MOINS 70% DE NON NULL

Liste_à_drop = []
for features in Fichier_Country_Series.columns:
    if Fichier_Country_Series[features].isnull().sum()>0.3*Fichier_Country_Series.shape[0]:
        Liste_à_drop.append(features)
print(Liste_à_drop)
Fichier_Country_Series=Fichier_Country_Series.drop(Liste_à_drop,axis=1)
Fichier_Country_Series.info()

['Unnamed: 3']
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 613 entries, 0 to 612
Data columns (total 3 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   CountryCode  613 non-null    object
 1   SeriesCode   613 non-null    object
 2   DESCRIPTION  613 non-null    object
dtypes: object(3)
memory usage: 14.5+ KB


# ***<u>PASSONS AU PROCHAIN FICHIER NOMME EdStatsData</u>***



In [316]:
# definition du path
path=r'C:\Users\lenovo\Desktop\OpenClassrooms\PROJET_OC\PROJET_2\RESSOURCES\Projet+Python_Dataset_Edstats_csv\EdStatsData.csv'

In [317]:
# CREATION DU DATAFRAME CORRESPONDANT
Fichier_Data = pd.read_csv(path)

In [318]:
# ETUDE DES 5 PREMIERES LIGNES DU DATAFRAME
Fichier_Data.head(2)

Country Name Country Code  \
0   Arab World          ARB   
1   Arab World          ARB   

                                      Indicator Name Indicator Code  1970  \
0  Adjusted net enrolment rate, lower secondary, ...     UIS.NERA.2   NaN   
1  Adjusted net enrolment rate, lower secondary, ...   UIS.NERA.2.F   NaN   

   1971  1972  1973  1974  1975  ...  2060  2065  2070  2075  2080  2085  \
0   NaN   NaN   NaN   NaN   NaN  ...   NaN   NaN   NaN   NaN   NaN   NaN   
1   NaN   NaN   NaN   NaN   NaN  ...   NaN   NaN   NaN   NaN   NaN   NaN   

   2090  2095  2100  Unnamed: 69  
0   NaN   NaN   NaN          NaN  
1   NaN   NaN   NaN          NaN  

[2 rows x 70 columns]

DE PRIME ABORD ON PEUT CONSTATER QUE CE FICHIER EST TRIE PAR ORDRE ALPHABETIQUE DE COUNTRY NAME, CEPENDANT ICI IL SEMBLE PLUTOT S'AGIR DE GROUPE GEOGRAPHIQUE COMME Arab World (Monde Arabe) OU ENCORE East Asia & Pacific (Espace Asie Pacifique). IL FAUDRA EXPLORER PLUS EN PROFONDEUR CES RENSEIGNEMENTS

IL COMPORTE 70 COLONNES COMME LE COUNTRY CODE QUI SERT D'IDENTIFIANT

DEUX COLONNES NOMMEES Country Name ET Country Code QUI REPRESENTENT EN FAIT LES CODE DE REGIONS GEOGRAPHIOQUES AINSI QUE LEUR VERSIONS ABREGEES (EX : Arab World -> ARB)

DEUX COLONNES NOMMEES Indicator Name ET Indicator Code QUI SONT L'INDICATEUR ETUDIE AINSI QUE SON ABREVIATION (EX : Adujstment... -> UIS.NERA.2)

LA DERNIERE NOMMEE Unnamed: 69 CERTAINEMENT VIDE COMME SUR LES FICHIERS PRECEDENT

QUANT AUX 65 COLONNES RESTANTES ELLES REPRESENTENT TOUTES UNE ANNE DIFFERENTES CONTENANT LES VALEURS CONNUES POUR L'INDICATEUR PRECISE DANS LA COLONNE Indicator Name A LA DATE INDIQUEE PAR LA COLONNE. LES ANNEES AVANCENT D'UN PAS DE 1 DE 1970 A 2017 INCLUS PUIS AVANCENT D'UN PAS DE 5 DE 2020 A 2100 INCLUS.




In [319]:
# RENTRONS PLUS EN DETAIL DANS L'EXPLORATION DE CE DATAFRAME
Fichier_Data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 886930 entries, 0 to 886929
Data columns (total 70 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   Country Name    886930 non-null  object 
 1   Country Code    886930 non-null  object 
 2   Indicator Name  886930 non-null  object 
 3   Indicator Code  886930 non-null  object 
 4   1970            72288 non-null   float64
 5   1971            35537 non-null   float64
 6   1972            35619 non-null   float64
 7   1973            35545 non-null   float64
 8   1974            35730 non-null   float64
 9   1975            87306 non-null   float64
 10  1976            37483 non-null   float64
 11  1977            37574 non-null   float64
 12  1978            37576 non-null   float64
 13  1979            36809 non-null   float64
 14  1980            89122 non-null   float64
 15  1981            38777 non-null   float64
 16  1982            37511 non-null   float64
 17  1983      

In [320]:
Fichier_Data.loc[Fichier_Data['Country Code']=='ABW']

Country Name Country Code  \
124610        Aruba          ABW   
124611        Aruba          ABW   
124612        Aruba          ABW   
124613        Aruba          ABW   
124614        Aruba          ABW   
...             ...          ...   
128270        Aruba          ABW   
128271        Aruba          ABW   
128272        Aruba          ABW   
128273        Aruba          ABW   
128274        Aruba          ABW   

                                           Indicator Name  \
124610  Adjusted net enrolment rate, lower secondary, ...   
124611  Adjusted net enrolment rate, lower secondary, ...   
124612  Adjusted net enrolment rate, lower secondary, ...   
124613  Adjusted net enrolment rate, lower secondary, ...   
124614  Adjusted net enrolment rate, primary, both sex...   
...                                                   ...   
128270  Youth illiterate population, 15-24 years, male...   
128271  Youth literacy rate, population 15-24 years, b...   
128272  Youth literacy rate, population 15-24 years, f...   
128273  Youth literacy rate, population 15-24 years, g...   
128274  Youth literacy rate, population 15-24 years, m...   

              Indicator Code  1970  1971  1972  1973  1974  1975  ...  2060  \
124610            UIS.NERA.2   NaN   NaN   NaN   NaN   NaN   NaN  ...   NaN   
124611          UIS.NERA.2.F   NaN   NaN   NaN   NaN   NaN   NaN  ...   NaN   
124612        UIS.NERA.2.GPI   NaN   NaN   NaN   NaN   NaN   NaN  ...   NaN   
124613          UIS.NERA.2.M   NaN   NaN   NaN   NaN   NaN   NaN  ...   NaN   
124614           SE.PRM.TENR   NaN   NaN   NaN   NaN   NaN   NaN  ...   NaN   
...                      ...   ...   ...   ...   ...   ...   ...  ...   ...   
128270      UIS.LP.AG15T24.M   NaN   NaN   NaN   NaN   NaN   NaN  ...   NaN   
128271     SE.ADT.1524.LT.ZS   NaN   NaN   NaN   NaN   NaN   NaN  ...   NaN   
128272  SE.ADT.1524.LT.FE.ZS   NaN   NaN   NaN   NaN   NaN   NaN  ...   NaN   
128273  SE.ADT.1524.LT.FM.ZS   NaN   NaN   NaN   NaN   NaN   NaN  ...   NaN   
128274  SE.ADT.1524.LT.MA.ZS   NaN   NaN   NaN   NaN   NaN   NaN  ...   NaN   

        2065  2070  2075  2080  2085  2090  2095  2100  Unnamed: 69  
124610   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
124611   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
124612   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
124613   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
124614   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
...      ...   ...   ...   ...   ...   ...   ...   ...          ...  
128270   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
128271   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
128272   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
128273   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
128274   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  

[3665 rows x 70 columns]


CE FICHIER CONTIENT  886930 LIGNES NUMEROTEES DE 0 A 886929 ET 70 COLONNES NUMEROTEES DE 0 A 69. IL COMPORTE 66 COLONNE DE TYPES FLOAT (NOMBRES DECIMAUX) ET 4 DE TYPE OBJECT (CHAINE DE CARACTERES). 



ON CONSTATE QUE LA COLONNE N° 69 (Unnamed: 69) EST ENTIEREMENT VIDE, EN EFFET ELLE CONTIENT 0 NON-NULL. LA COLONNE N°51 REPRESENTANT L'ANNEE 2017 EST TRES PEU RENSEIGNEE (SEULEMENT 143 NON-NULL).LES ANNEES TERMINANT PAR 0 OU 5 ENTRE 1970 ET 1990 INCLUS SONT MIEUX RENSEIGNEES QUE LES AUTRES SUR LA MEME PERIODE. CE PHENOMENE CE REPETE DE 1990 A 2000 BIEN QUE SUR CETTE PERIODE LES "FAIBLES" RENSEIGNEMENTS ONT OTUT DE MEME DOUBLES EN VALEUR ABSOLUE. A PARTIR DES ANNEES 1999-2000 JUSQU'EN 2015 LES RENSEIGNEMENTS VONT DE 120K A 200K VOIR 240K. 2016 ET 2017 ETANT TRES MAL RENSEIGNEES LES COLONNES N°52 JUSQU'A 68 c.à.d. DE L'ANNEE 2020 JUSQU'A L'ANNEE 2100 COMPORTE TOUTES 51436 VALEURS NON NULLES (CERTAINEMENT DES PROJECTIONS DE VALEURS)


In [321]:
Fichier_Data['Country Name'].unique()

array(['Arab World', 'East Asia & Pacific',
       'East Asia & Pacific (excluding high income)', 'Euro area',
       'Europe & Central Asia',
       'Europe & Central Asia (excluding high income)', 'European Union',
       'Heavily indebted poor countries (HIPC)', 'High income',
       'Latin America & Caribbean',
       'Latin America & Caribbean (excluding high income)',
       'Least developed countries: UN classification',
       'Low & middle income', 'Low income', 'Lower middle income',
       'Middle East & North Africa',
       'Middle East & North Africa (excluding high income)',
       'Middle income', 'North America', 'OECD members', 'South Asia',
       'Sub-Saharan Africa', 'Sub-Saharan Africa (excluding high income)',
       'Upper middle income', 'World', 'Afghanistan', 'Albania',
       'Algeria', 'American Samoa', 'Andorra', 'Angola',
       'Antigua and Barbuda', 'Argentina', 'Armenia', 'Aruba',
       'Australia', 'Austria', 'Azerbaijan', 'Bahamas, The', 'Bahrain',


ON CONSTATE QUE DANS CETTE TABLE LE COUNTRY CODE DESIGNE DES PAYS MAIS AUSSI DES ZONES REGIONALES (COMME ASIE DE L'EST ET PACIFIQUE (AVEC LES LOW INCOME INCLUS OU NON), PAYS DE L'OCDE, ZONE EURO...) VOIR LA TOTALITE DU MONDE.


In [322]:
# ON VA DROP TOUTES LES COLONNES QUI N'ONT PAS AU MOINS 70% DE NON NULL

Liste_à_drop = []
for features in Fichier_Data.columns:
    if Fichier_Data[features].isnull().sum()>0.95*Fichier_Data.shape[0]:
        Liste_à_drop.append(features)
print(Liste_à_drop)
Fichier_Data=Fichier_Data.drop(Liste_à_drop,axis=1)
Fichier_Data.info()

['1971', '1972', '1973', '1974', '1976', '1977', '1978', '1979', '1981', '1982', '1983', '1984', '1986', '1987', '1988', '1989', '2016', '2017', 'Unnamed: 69']
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 886930 entries, 0 to 886929
Data columns (total 51 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   Country Name    886930 non-null  object 
 1   Country Code    886930 non-null  object 
 2   Indicator Name  886930 non-null  object 
 3   Indicator Code  886930 non-null  object 
 4   1970            72288 non-null   float64
 5   1975            87306 non-null   float64
 6   1980            89122 non-null   float64
 7   1985            90296 non-null   float64
 8   1990            124405 non-null  float64
 9   1991            74437 non-null   float64
 10  1992            75543 non-null   float64
 11  1993            75793 non-null   float64
 12  1994            77462 non-null   float64
 13  1995            131361 non-null  f

In [323]:
# ON VA MAINTENANT DROP LES LIGNES TROP PEU RENSEIGNEES (C.A.D. QUI ONT SEULEMENT LES QUATRES PREMIERES COLONNES COMPLETEES)

Fichier_Data=Fichier_Data.dropna(thresh=22)
# J'AI CHOISI UNE LIMITE A AU MOINS 22 NON NULL DANS LA LIGNE AFIN D'INCLURE LES LIGNES QUI ONT AU MOINS UNE DONNEE SUPPLEMENTAIRE
# QUE LES 4 PREMIERES(QUI NE MANQUENT NULLE PART) ET LES 17 DERNIERES QUI SONT DES PROJECTIONS
Fichier_Data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 45580 entries, 4 to 886252
Data columns (total 51 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Country Name    45580 non-null  object 
 1   Country Code    45580 non-null  object 
 2   Indicator Name  45580 non-null  object 
 3   Indicator Code  45580 non-null  object 
 4   1970            14822 non-null  float64
 5   1975            19005 non-null  float64
 6   1980            19562 non-null  float64
 7   1985            20271 non-null  float64
 8   1990            45321 non-null  float64
 9   1991            45453 non-null  float64
 10  1992            45391 non-null  float64
 11  1993            45420 non-null  float64
 12  1994            45408 non-null  float64
 13  1995            45454 non-null  float64
 14  1996            45430 non-null  float64
 15  1997            44682 non-null  float64
 16  1998            44303 non-null  float64
 17  1999            45515 non-null  flo

In [324]:
Fichier_Data.head()

Country Name Country Code  \
4    Arab World          ARB   
5    Arab World          ARB   
6    Arab World          ARB   
7    Arab World          ARB   
12   Arab World          ARB   

                                       Indicator Name  Indicator Code  \
4   Adjusted net enrolment rate, primary, both sex...     SE.PRM.TENR   
5    Adjusted net enrolment rate, primary, female (%)  SE.PRM.TENR.FE   
6   Adjusted net enrolment rate, primary, gender p...  UIS.NERA.1.GPI   
7      Adjusted net enrolment rate, primary, male (%)  SE.PRM.TENR.MA   
12  Adjusted net intake rate to Grade 1 of primary...      UIS.NIRA.1   

         1970       1975       1980       1985       1990       1991  ...  \
4   54.822121  59.365540  65.617767  69.033211  71.995819  72.602837  ...   
5   43.351101  48.363892  56.486790  61.520870  65.827492  66.797028  ...   
6    0.658570   0.691760   0.759440   0.806770   0.844660   0.854320  ...   
7   65.826233  69.914551  74.379982  76.255493  77.933853  78.187492  ...   
12  52.448921  54.865627  62.559566  64.613091  67.080444  67.508705  ...   

    2055  2060  2065  2070  2075  2080  2085  2090  2095  2100  
4    NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
5    NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
6    NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
7    NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  
12   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN  

[5 rows x 51 columns]

In [325]:
Fichier_Data.loc[Fichier_Data['Country Code']=='ABW']

Country Name Country Code  \
126291        Aruba          ABW   
126293        Aruba          ABW   
126294        Aruba          ABW   
126917        Aruba          ABW   
126921        Aruba          ABW   
...             ...          ...   
127592        Aruba          ABW   
127594        Aruba          ABW   
127595        Aruba          ABW   
127596        Aruba          ABW   
127597        Aruba          ABW   

                                           Indicator Name  Indicator Code  \
126291  Official entrance age to lower secondary educa...     SE.SEC.AGES   
126293  Official entrance age to pre-primary education...     UIS.THAGE.0   
126294  Official entrance age to primary education (ye...     SE.PRM.AGES   
126917                       Population growth (annual %)     SP.POP.GROW   
126921  Population of the official age for lower secon...  SP.SEC.LTOT.IN   
...                                                   ...             ...   
127592  Theoretical duration of lower secondary educat...  SE.SEC.DURS.LO   
127594  Theoretical duration of pre-primary education ...     UIS.THDUR.0   
127595  Theoretical duration of primary education (years)     SE.PRM.DURS   
127596  Theoretical duration of secondary education (y...     SE.SEC.DURS   
127597  Theoretical duration of upper secondary educat...  SE.SEC.DURS.UP   

               1970         1975         1980         1985         1990  \
126291    12.000000    12.000000    12.000000    12.000000    12.000000   
126293     4.000000     4.000000     4.000000     4.000000     4.000000   
126294     6.000000     6.000000     6.000000     6.000000     6.000000   
126917     0.572211     0.212898     0.193211     0.301918     1.813641   
126921  2895.000000  2925.000000  2515.000000  1872.000000  1734.000000   
...             ...          ...          ...          ...          ...   
127592     2.000000     2.000000     2.000000     2.000000     2.000000   
127594     2.000000     2.000000     2.000000     2.000000     2.000000   
127595     6.000000     6.000000     6.000000     6.000000     6.000000   
127596     5.000000     5.000000     5.000000     5.000000     5.000000   
127597     3.000000     3.000000     3.000000     3.000000     3.000000   

               1991  ...  2055  2060  2065  2070  2075  2080  2085  2090  \
126291    12.000000  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
126293     4.000000  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
126294     6.000000  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
126917     3.902018  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
126921  1804.000000  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
...             ...  ...   ...   ...   ...   ...   ...   ...   ...   ...   
127592     2.000000  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
127594     2.000000  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
127595     6.000000  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
127596     5.000000  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   
127597     3.000000  ...   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   

        2095  2100  
126291   NaN   NaN  
126293   NaN   NaN  
126294   NaN   NaN  
126917   NaN   NaN  
126921   NaN   NaN  
...      ...   ...  
127592   NaN   NaN  
127594   NaN   NaN  
127595   NaN   NaN  
127596   NaN   NaN  
127597   NaN   NaN  

[177 rows x 51 columns]

# ***<u>PASSONS AU PROCHAIN FICHIER NOMME EdStatsFootNote</u>***

In [326]:
#definition du path
path=r'C:\Users\lenovo\Desktop\OpenClassrooms\PROJET_OC\PROJET_2\RESSOURCES\Projet+Python_Dataset_Edstats_csv\EdStatsFootNote.csv'

In [327]:
# CREATION DU DATAFRAME CORRESPONDANT
Fichier_FootNote = pd.read_csv(path)

In [328]:
# ETUDE DES 5 PREMIERES LIGNES DU DATAFRAME
Fichier_FootNote.head(5)

CountryCode      SeriesCode    Year          DESCRIPTION  Unnamed: 4
0         ABW  SE.PRE.ENRL.FE  YR2001  Country estimation.         NaN
1         ABW  SE.TER.TCHR.FE  YR2005  Country estimation.         NaN
2         ABW  SE.PRE.TCHR.FE  YR2000  Country estimation.         NaN
3         ABW  SE.SEC.ENRL.GC  YR2004  Country estimation.         NaN
4         ABW     SE.PRE.TCHR  YR2006  Country estimation.         NaN


DE PRIME ABORD ON PEUT CONSTATER QUE CE FICHIER EST TRIE PAR ORDRE ALPHABETIQUE DE COUNTRY CODE, ABW POUR ARUBA COMME ON A PU LE VOIR DANS LE 1ER FICHIER. IL FAUDRA EXPLORER PLUS EN PROFONDEUR CES RENSEIGNEMENTS

IL COMPORTE 5 COLONNES COMME LE COUNTRY CODE QUI SERT D'IDENTIFIANT

UNE COLONNE CountryCode REFERENCANT LA ZONE GEOGRAPHIQUE

UNE COLONNE NOMMEE SeriesCode REFERENCANT LE CODE DE LA SERIE DE DONNEES

UNE COLONNE Year RENSEIGNANT L'ANNEE A LAQUELLE SE RAPPORTE LA SERIE

UNE COLONNE DESCRIPTION EXPLIQUANT SI LA DONNEE EST UNE ESTIMATION OU UNE DONNEE COLLECTEE

LA DERNIERE NOMMEE Unnamed: 4 CERTAINEMENT VIDE COMME SUR LES FICHIERS PRECEDENT


In [329]:
# RENTRONS PLUS EN DETAIL DANS L'EXPLORATION DE CE DATAFRAME
Fichier_FootNote.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 643638 entries, 0 to 643637
Data columns (total 5 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   CountryCode  643638 non-null  object 
 1   SeriesCode   643638 non-null  object 
 2   Year         643638 non-null  object 
 3   DESCRIPTION  643638 non-null  object 
 4   Unnamed: 4   0 non-null       float64
dtypes: float64(1), object(4)
memory usage: 24.6+ MB



CE FICHIER CONTIENT  643638 LIGNES NUMEROTEES DE 0 A 643637 ET 5 COLONNES NUMEROTEES DE 0 A 4. IL COMPORTE 1 COLONNE DE TYPES FLOAT (NOMBRES DECIMAUX) ET 4 DE TYPE OBJECT (CHAINE DE CARACTERES). 

ON CONSTATE QUE LA COLONNE N° 4 (Unnamed: 4) EST ENTIEREMENT VIDE, EN EFFET ELLE CONTIENT 0 NON-NULL. QUANT AUX AUTRES COLONNE ELLES SONT TOUTES COMPLETES


In [330]:
# ON VA DROP TOUTES LES COLONNES QUI N'ONT PAS AU MOINS 70% DE NON NULL

Liste_à_drop = []
for features in Fichier_FootNote.columns:
    if Fichier_FootNote[features].isnull().sum()>0.3*Fichier_FootNote.shape[0]:
        Liste_à_drop.append(features)
print(Liste_à_drop)
Fichier_FootNote=Fichier_FootNote.drop(Liste_à_drop,axis=1)
Fichier_FootNote.info()

['Unnamed: 4']
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 643638 entries, 0 to 643637
Data columns (total 4 columns):
 #   Column       Non-Null Count   Dtype 
---  ------       --------------   ----- 
 0   CountryCode  643638 non-null  object
 1   SeriesCode   643638 non-null  object
 2   Year         643638 non-null  object
 3   DESCRIPTION  643638 non-null  object
dtypes: object(4)
memory usage: 19.6+ MB


# ***<u>PASSONS AU PROCHAIN FICHIER NOMME EdStatsSeries</u>***

In [331]:
#definition du path
path=r'C:\Users\lenovo\Desktop\OpenClassrooms\PROJET_OC\PROJET_2\RESSOURCES\Projet+Python_Dataset_Edstats_csv\EdStatsSeries.csv'

In [332]:
# CREATION DU DATAFRAME CORRESPONDANT
Fichier_Series = pd.read_csv(path)

In [333]:
# ETUDE DES 5 PREMIERES LIGNES DU DATAFRAME
Fichier_Series.head(2)

Series Code       Topic  \
0  BAR.NOED.1519.FE.ZS  Attainment   
1     BAR.NOED.1519.ZS  Attainment   

                                      Indicator Name  \
0  Barro-Lee: Percentage of female population age...   
1  Barro-Lee: Percentage of population age 15-19 ...   

                                    Short definition  \
0  Percentage of female population age 15-19 with...   
1  Percentage of population age 15-19 with no edu...   

                                     Long definition  Unit of measure  \
0  Percentage of female population age 15-19 with...              NaN   
1  Percentage of population age 15-19 with no edu...              NaN   

  Periodicity Base Period Other notes Aggregation method  ...  \
0         NaN         NaN         NaN                NaN  ...   
1         NaN         NaN         NaN                NaN  ...   

  Notes from original source  General comments  \
0                        NaN               NaN   
1                        NaN               NaN   

                                              Source  \
0  Robert J. Barro and Jong-Wha Lee: http://www.b...   
1  Robert J. Barro and Jong-Wha Lee: http://www.b...   

  Statistical concept and methodology Development relevance  \
0                                 NaN                   NaN   
1                                 NaN                   NaN   

  Related source links Other web links  Related indicators  License Type  \
0                  NaN             NaN                 NaN           NaN   
1                  NaN             NaN                 NaN           NaN   

   Unnamed: 20  
0          NaN  
1          NaN  

[2 rows x 21 columns]


DE PRIME ABORD ON PEUT CONSTATER QUE CE FICHIER EST TRIE PAR ORDRE ALPHABETIQUE DE Series Code, IL SEMBLERAIT QUE DANS LE NOM SE TROUVE LA TRNACHE D'AGE ETUDIE PAR EXEMPLE 1519 POUR LES 15-19 ANS. IL FAUDRA EXPLORER PLUS EN PROFONDEUR CES RENSEIGNEMENTS

L'ENSEMBLE DES COLONNES A POUR BUT DE RENSEIGNER LA MANIERE DONT ONT ETE COLLECTEES LES DONNEES, LES SOURCES UTILISEES, LES UNITES DE MESURES...

In [334]:
# RENTRONS PLUS EN DETAIL DANS L'EXPLORATION DE CE DATAFRAME
Fichier_Series.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3665 entries, 0 to 3664
Data columns (total 21 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   Series Code                          3665 non-null   object 
 1   Topic                                3665 non-null   object 
 2   Indicator Name                       3665 non-null   object 
 3   Short definition                     2156 non-null   object 
 4   Long definition                      3665 non-null   object 
 5   Unit of measure                      0 non-null      float64
 6   Periodicity                          99 non-null     object 
 7   Base Period                          314 non-null    object 
 8   Other notes                          552 non-null    object 
 9   Aggregation method                   47 non-null     object 
 10  Limitations and exceptions           14 non-null     object 
 11  Notes from original source    


CE FICHIER CONTIENT  3665 LIGNES NUMEROTEES DE 0 A 3664 ET 21 COLONNES NUMEROTEES DE 0 A 20. IL COMPORTE 6 COLONNE DE TYPES FLOAT (NOMBRES DECIMAUX) ET 15 DE TYPE OBJECT (CHAINE DE CARACTERES). 
A NOTER QUE TOUTES LES COLONNES DE TYPES FLOAT SONT ENTIEREMENT VIDES


LES COLONNES N° 0, 1, 2, 4, 13 SONT COMPLETES. LA COLONNE N°3 Short Definition EST QUANT A ELLE COMPLETES A ENVIRON 60%. QUANT AUX AUTRES ELLES NE DEPASSENT PAS LES 552 RENSEIGNEMENTS (ENVIRON 15%). ENFIN LES N° 5, 11, 17, 18, 19, 20 SONT ENTIEREMENT VIDES

In [335]:
# ON VA DROP TOUTES LES COLONNES QUI N'ONT PAS AU MOINS 70% DE NON NULL

Liste_à_drop = []
for features in Fichier_Series.columns:
    if Fichier_Series[features].isnull().sum()>0.3*Fichier_Series.shape[0]:
        Liste_à_drop.append(features)
print(Liste_à_drop)
Fichier_Series=Fichier_Series.drop(Liste_à_drop,axis=1)
Fichier_Series.info()

['Short definition', 'Unit of measure', 'Periodicity', 'Base Period', 'Other notes', 'Aggregation method', 'Limitations and exceptions', 'Notes from original source', 'General comments', 'Statistical concept and methodology', 'Development relevance', 'Related source links', 'Other web links', 'Related indicators', 'License Type', 'Unnamed: 20']
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3665 entries, 0 to 3664
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Series Code      3665 non-null   object
 1   Topic            3665 non-null   object
 2   Indicator Name   3665 non-null   object
 3   Long definition  3665 non-null   object
 4   Source           3665 non-null   object
dtypes: object(5)
memory usage: 143.3+ KB


# ***<u>FIN ETUDE PRELIMINAIRE</u>***

REPRISE DE L'ANALYSE

LE FICHIER "Fichier_Series" NE SEMBLE PAS PERTINENT POUR L'ANALYSE (IL DETAILLE EN FAIT LA METHODOLOGIE APPLIQUEE POUR RECUEUILLIR L'ENSEMBLE DES DONNEES)

DE MEME POUR "Fichier_Footnote" AINSI QUE "Fichier_Country_Series" QUI LA AUSSI SONT DES FICHIERS DE METHODOLOGIE.